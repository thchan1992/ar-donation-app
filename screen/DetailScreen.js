import React from 'react';
import {View, Text, Image, StyleSheet, ScrollView} from 'react-native';
import {image} from '../util/styling';
import {horizontalLine} from '../util/styling';
import PrimaryButton from '../component/PrimaryButton';

const DetailScreen = ({route, navigation}) => {
  const {charity} = route.params;
  return (
    <View style={styles.container}>
      <View style={[{alignItems: 'center'}, horizontalLine.horizontalLine]}>
        <Text style={styles.text}>{charity.name}</Text>
        <PrimaryButton
          text={'Donate'}
          onPress={() => {
            navigation.navigate('donate', {charity: charity});
          }}
        />
      </View>
      <ScrollView>
        <Text style={styles.infoText}>{charity.info}</Text>
      </ScrollView>
    </View>
  );
};

export default DetailScreen;

const styles = StyleSheet.create({
  image: {width: 200, height: 200},
  container: {flex: 1, alignItems: 'center', paddingTop: 10},
  text: {
    alignSelf: 'center',
    padding: 3,
    fontWeight: 'bold',
    textTransform: 'uppercase',
    fontSize: 20,
    color: 'white',
  },
  infoText: {
    padding: 10,
    fontSize: 30,
    color: 'white',
  },
});
