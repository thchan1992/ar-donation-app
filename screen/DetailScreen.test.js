import React from 'react';
import renderer from 'react-test-renderer';
import DetailScreen from './DetailScreen';

describe('DetailScreen', () => {
  it('renders correctly', () => {
    const charity = {
      name: 'Test Charity',
      info: 'Test charity information.',
    };
    const navigation = {
      navigate: jest.fn(),
    };
    const route = {
      params: {
        charity,
      },
    };
    const tree = renderer
      .create(<DetailScreen route={route} navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
