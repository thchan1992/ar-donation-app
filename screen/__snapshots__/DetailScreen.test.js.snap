// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "borderBottomColor": "#FF304F",
          "borderBottomWidth": 0.5,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 3,
          "textTransform": "uppercase",
        }
      }
    >
      Test Charity
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "bold",
              "padding": 16,
            },
            Object {
              "alignSelf": "center",
              "fontSize": 30,
              "fontWeight": "bold",
              "padding": 16,
            },
          ]
        }
      >
        Donate
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "padding": 10,
          }
        }
      >
        Test charity information.
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
